### ISSUE TITLE
0.11.5 is a breaking change

### ISSUE THREAD

**Issue by CryZe** [emberian/evdev#66]
You accidentally released a breaking change as part of 0.11.5 (enumerate), the release should be yanked.

**Comment by coolreader18** [comment]
Ah, thanks, sorry about that.

**Comment by coolreader18** [comment]
I guess I'll release a 0.11.6 and then a 0.12

**Comment by CryZe** [comment]
It looks like 0.11.6 contains the breaking change again.

**Comment by jeff-hiner** [comment]
What's the breaking change?

**Comment by CryZe** [comment]
Someone reported this to me:
```
rror[E0599]: no method named `supported_events` found for reference `&(PathBuf, Device)` in the current scope
   --> /home/dagit/.cargo/registry/src/github.com-1ecc6299db9ec823/livesplit-hotkey-0.6.0/src/linux/mod.rs:281:27
    |
281 |             .filter(|d| d.supported_events().contains(EventType::KEY))
    |                           ^^^^^^^^^^^^^^^^ method not found in `&(PathBuf, Device)`

error[E0277]: a value of type `Vec<Device>` cannot be built from an iterator over elements of type `(PathBuf, Device)`
   --> /home/dagit/.cargo/registry/src/github.com-1ecc6299db9ec823/livesplit-hotkey-0.6.0/src/linux/mod.rs:282:14
    |
282 |             .collect();
    |              ^^^^^^^ value of type `Vec<Device>` cannot be built from `std::iter::Iterator<Item=(PathBuf, Device)>`
    |
    = help: the trait `FromIterator<(PathBuf, Device)>` is not implemented for `Vec<Device>`
    = help: the trait `FromIterator<T>` is implemented for `Vec<T>`
note: required by a bound in `collect`
```

I feel like it's the same as before? Maybe not.

**Comment by jeff-hiner** [comment]
@coolreader18 it appears the breaking change was in `fn enumerate`, changed in commit 9f63c9a0ebd32575e17433eb7ddfab8a22f1985b.

The iterator `EnumerateDevices` now returns `(PathBuf, RawDevice)` instead of just the device.

**Comment by coolreader18** [comment]
Oh, right, I forgot about that; I meant to make any new releases in a 0.12.x line but didn't remember when releasing 0.11.6. I guess I could release 0.11.7 with just that change reverted, and then 0.12 from there on out?

**Comment by CryZe** [comment]
Yeah that sounds good.

**Comment by Shinyzenith** [comment]
Could `0.11.6` be yanked? it contains the same breaking release as `0.11.5`.
Last time I couldn't figure out what broke my code as it was a minor version bump...

**Comment by coolreader18** [comment]
Ok! 0.11.6 yanked, 0.11.7 published, 0.12.0 published! Sorry for all the confusion, everyone.

**Comment by Shinyzenith** [comment]
Thanks!
